jormungandr/src/utils/task.rs

```diff
@@ -183,7 +183,19 @@ impl Services {
             executor: executor,
         };
 
-        runtime.spawn(f(future_service_info));
+        use std::panic::AssertUnwindSafe;
+
+        let future = AssertUnwindSafe(f(future_service_info))
```

Comment:
I would not assert it for a generic closure, but add `UnwindSafe` bound for the `F` closure type, and address unwind safety in tasks on a case-by-case basis, if required by the compiler.

Reply:
tried that and that is 2 years of software development to resolve it that way

Reply:
Not sure what is required besides, in the worst case, shifting the `AssertUnwindSafe` wrapper to every use site for the few tasks we have, to leave audit markers for re-evaluation later if this causes problems.
But let's move it forward.

Reply:
I have created an issue to track that down at some point